Add autoClose flag

povesteam opened the pull request:
<img src="https://user-images.githubusercontent.com/506532/60768202-f07be480-a0ca-11e9-90ca-2d413f83c7de.png">

Add the `autoClose` prop to `<StyledWindowPortal />` component to close child window automatically when parent is closed (via window close, tab close, or refresh).

povesteam committed:
Add autoClose flag

Briggybros commented on src/StyledWindowPortal.tsx:
```diff
@@ -93,12 +94,26 @@ class StyledWindowPortal extends React.PureComponent<Props, State> {
-    if (!!this.state.externalWindow) this.state.externalWindow.close();
+    if (this.props.autoClose) {
+      window.removeEventListener("unload", this.closeExternalWindow);
+    }
```
Is there a case here where if the component is mounted with autoClose = true and is subsequently changed to false before unmounting then this removeEventListener is not called?

Removing the conditional here would mean the event listener is removed if it exists and does not harm if it does not.

Briggybros reviewed (changes requested):
See inline comment

povesteam committed:
Add and remove unload listener according to autoClose prop

povesteam commented:
@Briggybros Thanks for the good review!

Updated the code to add and remove the event listener according to the `autoClose` prop, to prevent overloading the document with unnecessary listeners.

![image](https://user-images.githubusercontent.com/506532/60901296-5f497100-a276-11e9-9f59-82cf5d022d58.png)

Also, added a checkbox in the test file, to inspect the listener add/remove process.

![image](https://user-images.githubusercontent.com/506532/60901462-af283800-a276-11e9-9821-9f2df6699519.png)

Briggybros commented:
@povesteam Thank you for the feature :)

povesteam commented:
Glad to improve something I'm using.

Thank you for the package!
